site/book/07-effective-customizations/01-single-value-replacement.md

```diff
@@ -0,0 +1,26 @@
+## Scenario
+
+I have a single value replacement in my package, I don’t want package consumers to look through all the yaml files to find the value I want them to set, it seems easier to just create a parameter for this value and have the user look at Kptfile for inputs.
+
+## Problems
+
+1. With popularity the single values inevitably expand to provide a facade to a large portion of the data defeating the purpose of minimizing the cognitive load.
```

Comment:
It would be pretty easy to find a helm chart to link to as an example. I've used the jenkins chart in the past. A search for "universal helm chart" found this:
https://github.com/OleksandrUA/universal-helm-charts/blob/master/chart-service/_common/templates/deployment.yaml

Reply:
Also, I call parameterizing every field the struct constructor pattern.

Reply:
https://github.com/jenkinsci/helm-charts/blob/main/charts/jenkins/templates/jenkins-controller-statefulset.yaml#L84

Reply:
I agree that it's not hard to find a chart that suffers from the parameter problem.  I think what @droot is asking for here is before and after so I'd have to create a feasible package for whatever it is that I showcase.  There is also a limit to the size of the example to show case the end result.  Maybe I can figure out how to just show code snippets here.